### PR TITLE
process document for jumbo PR

### PR DESCRIPTION
**Pull request by omgitsbillryan**
Connects: https://github.com/department-of-veterans-affairs/va.gov-team/issues/11630

A document that the Danger bot will link to when a "Jumbo" pull request is detected.
image

### CODE REVIEW

**Review by rianfowler — DISMISSED** [review]

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# So you have a Jumbo PR
+
+Find which of the following applies to you -
+
+- The changed lines of code in my PR is not an accurate measurement how much effort is required to review it 
+    - :point_right: [Change How LoC is Calculated](#change-how-loc-is-calculated)
+    - :point_right: [Appeal a Jumbo PR Denomination](#appeal-a-jumbo-pr-denomination)
+- There is no way my PR can be broken up into smaller pieces :point_right: [File for a Jumbo PR Override](#file-for-a-jumbo-pr-override)
+- I can break my PR up into smaller pieces :point_right: [Break PR Into Smaller Pieces](#break-pr-into-smaller-pieces)
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:9]:
```suggestion
- I can break my PR up into smaller pieces :point_right: [Break PR Into Smaller Pieces](#break-pr-into-smaller-pieces)
- The changed lines of code in my PR is not an accurate measurement how much effort is required to review it 
    - :point_right: [Change How LoC is Calculated](#change-how-loc-is-calculated)
    - :point_right: [Appeal a Jumbo PR Denomination](#appeal-a-jumbo-pr-denomination)
- There is no way my PR can be broken up into smaller pieces :point_right: [File for a Jumbo PR Override](#file-for-a-jumbo-pr-override)
```

Comment (gunsch) [reply platform/working-with-vsp/jumbo-pull-request.md:9]: suggest putting "break up my PR" as the first option ;)

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+## Change How LoC is Calculated
+### Example
+You added a large test file (`.csv`) that is no included in the [exclusions list](#jumbo-pr-calculation-rationale)
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:14]:
```suggestion
You added a large test file (`.csv`) that is not included in the [exclusions list](#jumbo-pr-calculation-rationale)
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+### What to Do
+1. Open a separate PR that changes how the Danger Bot calculates total LoC
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:17]: link to where this is done in source?

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+### What to Expect
+The platform team will review that PR inside the normal 24 hour cycle. If that PR is approved, you may then merge to `master` and pull those changes into your Jumbo PR branch which should now pass the Danger Bot LoC check.
+
+
+## Appeal a Jumbo PR Denomination
+### Example
+Your PR is mostly repetitive rspec code that isnt very complex.
+
+### What to Do
+Having > `1000` LoC, doesn't always mean the review will be difficult to review. In his case
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:29]:
```suggestion
Having > `1000` LoC, doesn't always mean the review will be difficult to review. In this case:
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Your PR is mostly repetitive rspec code that isnt very complex.
+
+### What to Do
+Having > `1000` LoC, doesn't always mean the review will be difficult to review. In his case 
+
+1. Open a ticket in the `va.gov-team` repo
+2. Use the "Jumbo PR Appeal" Github issue template
+3. Prefix the title with "Jumbo PR Appeal:"
+4. Add the github label "Jumbo PR Appeal"
+5. Add a comment in your PR that links to the appeal github issue
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:35]:
```suggestion
2. Use the "Jumbo PR Appeal" GitHub issue template
3. Prefix the title with "Jumbo PR Appeal:"
4. Add the GitHub label "Jumbo PR Appeal"
5. Add a comment in your PR that links to the appeal GitHub issue
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+
+## Appeal a Jumbo PR Denomination
+### Example
+Your PR is mostly repetitive rspec code that isnt very complex.
+
+### What to Do
+Having > `1000` LoC, doesn't always mean the review will be difficult to review. In his case 
+
+1. Open a ticket in the `va.gov-team` repo
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:31]:
```suggestion
1. [Open a ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new/choose) in the `va.gov-team` repo
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+## File for a Jumbo PR Override
+
+### Example
+There is no possible way your PR can be broken up and must be reviewed as is.
+
+### What to Do
+
+1. create a doc explaining why? (TBD)
+2. The platform team will review that doc (1-2)
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:49]:
```suggestion
2. The platform team will review that doc (1-2 days)
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+3. Schedule an in-person meeting with the BE utility dev & BE team product owner for a more in depth look at the PR & feature
+4. PR will occur normally
+
+### What to Expect
+3-7 days. This includes
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:54]:
```suggestion
Expect this process to take 3-7 days total. This includes:
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+- in person meeting
+- longer back-and-forth comments on the PR
+
+
+## Break PR Into Smaller Pieces
+
+### What to Do
+1. Close the Jumbo PR
+2. Link back to the jumbo PR in any future smaller chunk PRs
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:64]:
```suggestion
2. Link back to the Jumbo PR in any future smaller chunk PRs
```

File: platform/working-with-vsp/jumbo-pull-request.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+We use lines of code (`LoC`) as a measurement for its simplicity & transparency. We ignore the following when calculating LoC:
+
+```
+EXCLUSIONS = ['Gemfile.lock', '.json', 'spec/fixtures/', '.txt', 'spec/support/vcr_cassettes/', 'app/swagger', '.csv']
+```
+:warning: [The `Dangerfile` for actual calculation](https://github.com/department-of-veterans-affairs/vets-api/blob/master/Dangerfile) should be considered the source of truth, not this document! :warning:
+
+You can expand the comment from the bot to see exactly how your LoC were calculated.
+[screen shot of expanded comment]
```
Comment (gunsch) [inline platform/working-with-vsp/jumbo-pull-request.md:85]:
```suggestion
We use lines of code (`LoC`) as a measurement for its simplicity & transparency. We ignore a number of files when calculating LoC, such as data files (`.json`, `.csv`), `Gemfile.lock`, and `vcr_cassettes`. See the [`vets-api Dangerfile`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/Dangerfile) for the full list of exclusions.

You can expand the comment from the bot to see exactly how your LoC were calculated.
[screen shot of expanded comment]
```

**Review by gunsch — COMMENTED** [review]
I'm fine with this policy-wise. Left a number of more editorial suggestions.

**Review by johnpaulashenfelter — CHANGES_REQUESTED** [review]
Started a chat with the tech leads -- TL;DR is this is way more process than we need. We're 100% in favor of updating the code review guidelines document with a Q/A about big PRs, and stressing (a) it will be automatically rejected by dangerbot (which is a review), and that (b) you'll need to work with us and it will take a while.

**Comment by johnpaulashenfelter** [comment]
@omgitsbillryan #12239

**Comment by omgitsbillryan** [comment]
I agree, this is a bit much for process. It's easy to get carried away when writing this stuff 😜  . My only intent with this PR was to give devs who get blocked by the danger-bot somewhere to go. If that "somewhere to go" is a link to a simple Q/A section of a doc, that's probably better 👍

**Pull request closed without merging**